format/index.md

```diff
@@ -52,761 +58,508 @@ called out below.
 
 ```text
-GET /comments?post=1&author=12
+GET /articles?include=author&fields[articles]=id,title&fields[people]=id,name
 ```
 
-This specification only supports filtering based upon strict matching.
-Additional filtering allowed by an API should be specified in its profile (see
-[Extending](/extending)).
-
-### Inclusion of Linked Resources <a href="#fetching-includes" id="fetching-includes" class="headerlink"></a>
-
-A server **MAY** choose to support returning compound documents that include
-both primary and linked resource objects.
-
-An endpoint **MAY** return resources linked to the primary resource(s) by
-default.
+If a client requests a restricted set of fields, an endpoint **MUST NOT** include other
+fields in the response.
 
-An endpoint **MAY** also support custom inclusion of linked resources based upon
-an `include` request parameter. This parameter should specify the path to one or
-more resources relative to the primary resource. If this parameter is used,
-**ONLY** the requested linked resources should be returned alongside the primary
-resource(s).
-
-For instance, comments could be requested with a post:
+### Sorting <a href="#fetching-sorting" id="fetching-sorting" class="headerlink"></a>
 
-```text
-GET /posts/1?include=comments
-```
+A server **MAY** choose to support requests to sort resource collections
+according to one or more criteria.
 
-In order to request resources linked to other resources, the dot-separated path
-of each relationship should be specified:
+An endpoint **MAY** support requests to sort the primary data with a `sort`
+query parameter.
 
 ```text
-GET /posts/1?include=comments.author
+GET /people?sort=+age
 ```
 
-Note: a request for `comments.author` should not automatically also include
-`comments` in the response (although comments will obviously need to be queried
-in order to fulfill the request for their authors).
-
-Multiple linked resources could be requested in a comma-separated list:
+An endpoint **MAY** support multiple sort criteria by allowing
+comma-separated (U+002C COMMA, ",") fields as the value for `sort`. Sort
+criteria should be applied in the order specified.
 
 ```text
-GET /posts/1?include=author,comments,comments.author
+GET /people?sort=+age,+name
 ```
 
-### Sparse Fieldsets <a href="#fetching-sparse-fieldsets" id="fetching-sparse-fieldsets" class="headerlink"></a>
+The sort order for each field **MUST** be specified with one of the
+following prefixes:
 
-A server **MAY** choose to support requests to return only specific fields in
-resource object.
+* Plus (U+002B PLUS SIGN, "+") to request an ascending sort order.
+* Minus (U+002D HYPHEN-MINUS, "-") to request a descending sort order.
 
-An endpoint **MAY** support requests that specify fields for the primary
-resource type with a `fields` parameter.
+> Note: By requiring a sort order prefix instead of allowing a default
+order, JSON API avoids setting requirements for the first character in field
+names. 
 
```

Comment:
That's a legitimate take on this - and I can't say that I haven't been on the fence here. We decided on this trade-off so that names would have no restrictions, and to keep sorting requirements only affecting implementations that choose to support sorting. In other words, we wanted to avoid having an optional aspect of the spec driving requirements in a very core part of the spec.